src/requests_gssapi/gssapi_.py

```diff
@@ -144,6 +152,37 @@ def generate_request_header(self, response, host, is_preemptive=False):
         if self.mutual_authentication != DISABLED:
             gssflags.append(gssapi.RequirementFlag.mutual_authentication)
 
+        gss_cb = None
+        if self.channel_bindings == "tls-server-end-point":
+            if is_preemptive:
+                log.warning("channel_bindings were requested, but are unavailable for opportunistic authentication")
+            # The 'connection' attribute on raw is a public urllib3 API
+            # and can be None if the connection has been released.
+            elif getattr(response.raw, "connection", None) and getattr(response.raw.connection, "sock", None):
+                try:
+                    # Defer import so it's not a hard dependency.
+                    from cryptography import x509
+
+                    sock = response.raw.connection.sock
+
+                    der_cert = sock.getpeercert(binary_form=True)
+                    cert = x509.load_der_x509_certificate(der_cert)
+                    hash = cert.signature_hash_algorithm
+                    cert_hash = cert.fingerprint(hash)
```

Comment:
Probably fine for now but IIRC SHA256 should be used unless it's an RSA key with something greater than 256 bits. For example an ECDSA certificate will still need to use SHA256 for the P384 or P521 curves. I'll need to double check on that though.

Reply:
Yes sorry I was mistaken, just looked back on some previous implementation I’ve done and the ECDSA certs match up with the hash algorithm. Probably best to keep it simple with what you’ve got until someone brings up some esoteric example later in the future :)